packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,398 @@
+# FileUpload Decisions <!-- omit in toc -->
+
+The `FileUpload` component is used to handle file attachments, including the drag-and-drop interaction. It can be used in both controlled and uncontrolled forms. Primarily, it is used to upload files to a server or to display a list of uploaded files.
+
+This document outlines the API details of the `FileUpload` component, encompassing their structural composition, functional attributes, and visual representation.
+
+<img width="100%" src="./single-upload-variant.png" alt="Single File Upload">
+<img width="100%" src="./multiple-upload-variant.png" alt="Multiple Files Upload">
+<img width="100%" src="./file-upload-item-variants.png" alt="FileUploadItem Variants ">
+
+- [Design](#design)
+- [Anatomy](#anatomy)
+- [`FileUpload` Props](#fileupload-props)
+- [Usage](#usage)
+  - [Submit Button File Upload (Uncontrolled)](#submit-button-file-upload-uncontrolled)
+  - [Direct Upload with Progress Display (Controlled)](#direct-upload-with-progress-display-controlled)
+- [Examples](#examples)
+  - [Uncontrolled Usage](#uncontrolled-usage)
+    - [Single File selection](#single-file-selection)
+    - [Multiple File selection:](#multiple-file-selection)
+  - [Controlled Usage](#controlled-usage)
+    - [Direct Upload with Progress Display](#direct-upload-with-progress-display)
+- [Accessibility](#accessibility)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=78150%3A2625&mode=design&t=bKQvPInRmAoeAAFV-1) to all variants of the `FileUpload` component.
+
+## Anatomy
+
+<img width="100%" src="./file-upload-anatomy.png" alt="Single File Upload">
+<img width="100%" src="./file-upload-item-anatomy.png" alt="Multiple Files Upload">
+
+## `FileUpload` Props
+
+```ts
+// Check the File type from MDN for more details: https://developer.mozilla.org/en-US/docs/Web/API/File
+interface BladeFile extends File {
+  /**
+   * The unique identifier of the file.
+   */
+  id: string;
+  /**
+   * The file's upload status.
+   */
+  status?: 'uploading' | 'success' | 'error';
+  /**
+   * The percentage of file upload completion.
+   */
+  percent?: number;
+  /**
+   * Text indicating an error state
+   */
+  errorText?: string;
+}
+
+type BladeFileList = BladeFile[];
+
+type FileUploadCommonProps = {
+  /**
+   * Position of the label relative to the file upload area. Desktop only prop. Default value on mobile will be 'top'
+   *
+   * @default 'top'
+   */
+  labelPosition?: 'top' | 'left';
+  /**
+   * Defines the selection behavior of the FileUpload component
+   */
+  selectionType?: 'single' | 'multiple';
```

Comment:
Is there a scope for more types? Can we convert it to a boolean prop? For example `isMultiple`.

Reply:
Keeping it consistent with other component's API. [Ref. ChipGroup's API](https://blade.razorpay.com/?path=/docs/components-chip-chipgroup--docs)